File: study/lua/src/app/scene/view/seasonShop/SeasonShopCell.lua
-- @Author panhoa
-- @Date 10.15.2018
-- @Role 

local ListViewCellBase = require("app.ui.ListViewCellBase")
local SeasonShopCell = class("SeasonShopCell", ListViewCellBase)
local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local ShopActiveDataHelper = require("app.utils.data.ShopActiveDataHelper")

function SeasonShopCell:ctor()
	local resource = {
		file = Path.getCSB("SeasonShopCell", "seasonShop"),
	}
	SeasonShopCell.super.ctor(self, resource)
end

function SeasonShopCell:onCreate()
	local size = self._resourceNode:getContentSize()
	self:setContentSize(size.width, size.height)
end

function SeasonShopCell:update(cellIndex, cellData)
	for index = 1, 2 do
        self["_itemInfo"..index]:setVisible(false)
    end
    
    local function updateItem(index, itemData)
        local itemInfo = self["_itemInfo"..index]
		if itemInfo then
			itemInfo:updateUI(itemData)
			itemInfo:setVisible(true)
			itemInfo:setTag(cellIndex*2 + index)
			itemInfo:setCallBack(handler(self,self._onBtnClick))
		end
    end

	for itemIndex, itemData in ipairs(cellData) do
        updateItem(itemIndex, itemData) 
     end
end

function SeasonShopCell:_onBtnClick(iconNameNode)
	local curSelectedPos = iconNameNode:getTag()
	self:dispatchCustomCallback(curSelectedPos)
end


return SeasonShopCell

File: study/ts/script/scene/view/seasonShop/SeasonShopCell.ts
const { ccclass, property } = cc._decorator;

import CommonShopItemCell from '../../../ui/component/CommonShopItemCell'
import { handler } from '../../../utils/handler';

@ccclass
export default class SeasonShopCell extends cc.Component {

    @property({ type: cc.Node, visible: true })
    _resourceNode: cc.Node = null;

    @property({ type: CommonShopItemCell, visible: true })
    _itemInfo: CommonShopItemCell = null;

    private _customCallback;
    private _index;
    public onLoad() {
        var size = this._resourceNode.getContentSize();
        this.node.setContentSize(size.width, size.height);
    }

    public updateUI(cellIndex, cellData) {
        this._index = cellIndex;
        this._itemInfo.updateUI(cellData);
        this._itemInfo.node.active = true;
        this._itemInfo.setCallBack(handler(this, this._onBtnClick));
    }

    private _onBtnClick(iconNameNode) {
        if(this._customCallback != null)
        {
            this._customCallback(this._index);
        }
    }

    public setCustomCallback(customCallback: Function) {
        this._customCallback = customCallback;
    }
}